src/libs/data-table-versions.js

```diff
@@ -0,0 +1,109 @@
+import _ from 'lodash/fp'
+import { useState } from 'react'
+import { parseGsUri } from 'src/components/data/data-utils'
+import { Ajax } from 'src/libs/ajax'
+import { getUser } from 'src/libs/auth'
+import { notify } from 'src/libs/notifications'
+import { useCancellation } from 'src/libs/react-utils'
+import * as Utils from 'src/libs/utils'
+
+
+export const dataTableVersionsRoot = '.data-table-versions'
+
+export const saveDataTableVersion = async (workspace, entityType, { description = null } = {}) => {
+  const { workspace: { namespace, name, googleProject, bucketName } } = workspace
+
+  const timestamp = (new Date()).getTime()
+  const versionName = `${entityType}.v${timestamp}`
+
+  const tsvContent = await Ajax().Workspaces.workspace(namespace, name).getEntitiesTsv(entityType)
+
+  const tsvFile = new File([tsvContent], versionName, { type: 'text/tab-separated-values' })
+  await Ajax().Buckets.upload(googleProject, bucketName, `${dataTableVersionsRoot}/${entityType}/`, tsvFile)
+
+  const objectName = `${dataTableVersionsRoot}/${entityType}/${versionName}`
+  const createdBy = getUser().email
+  await Ajax().Buckets.patch(googleProject, bucketName, objectName, {
+    metadata: {
+      createdBy,
+      entityType,
+      timestamp,
+      description
+    }
```

Comment:
I think this is just a smidge clearer in compact form:
```suggestion
    metadata: { createdBy, entityType, timestamp, description }
```
Your call.